### PR TITLE
test for _mce_set_direction

### PR DESCRIPTION
**Pull request by pbearne**
Trac ticket: https://core.trac.wordpress.org/ticket/57192

### CODE REVIEW

File: tests/phpunit/tests/functions/mceSetDirection.php [hunk 1]
```diff
@@ -0,0 +1,51 @@
+<?php
+/**
+ * Tests for _mce_set_direction function.
+ *
+ * @group functions.php
+ *
+ * @covers ::_mce_set_direction
+ */#
```
Comment (mukeshpanchal27) [inline tests/phpunit/tests/functions/mceSetDirection.php:8]:
```suggestion
 */
```

File: tests/phpunit/tests/functions/mceSetDirection.php [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+	 */
+	public function test__mce_set_direction() {
+		global $wp_locale;
+		$direction = $wp_locale->text_direction;
+
+		$mce_init = array(
+			'plugins'  => 'some_plugin_name',
+			'toolbar1' => 'some_button_name',
+		);
+		$this->assertEquals( $mce_init, _mce_set_direction( $mce_init ), ' set to ltr' );
```
Comment (mukeshpanchal27) [inline tests/phpunit/tests/functions/mceSetDirection.php]:
```suggestion
		$this->assertEquals( $mce_init, _mce_set_direction( $mce_init ), 'set to ltr' );
```

**Review by mukeshpanchal27 — COMMENTED** [review]
Thanks @pbearne Left some minor formatting feedback.

The message needs to update with more context.

File: tests/phpunit/tests/functions/mceSetDirection.php [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+
+	public function set_up() {
+		parent::set_up();
+		$this->locale = new WP_Locale();
+	}
+
+	/**
+	 * @ticket 57192
+	 */
+	public function test__mce_set_direction() {
```
Comment (costdev) [inline tests/phpunit/tests/functions/mceSetDirection.php]:
```suggestion
	public function test_mce_set_direction() {
```
A double `_` is not convention in the test suite, even for `_private` functions.

File: tests/phpunit/tests/functions/mceSetDirection.php (hunk 2 above)
Comment (costdev) [inline tests/phpunit/tests/functions/mceSetDirection.php]: As a string array, `assertSame()` should be used here.

File: tests/phpunit/tests/functions/mceSetDirection.php [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+			'directionality' => 'rtl',
+			'rtl_ui'         => true,
+		);
+		$this->assertEquals( $mce_init_expected, _mce_set_direction( $mce_init ), 'set to rtl' );
+
+		$mce_init = array(
+			'plugins'  => 'some_plugin_name,directionality',
+			'toolbar1' => 'some_button_name,ltr',
+		);
+		$this->assertEquals( $mce_init_expected, _mce_set_direction( $mce_init ), "checking we don't get to strings added" );
```
Comment (costdev) [inline tests/phpunit/tests/functions/mceSetDirection.php]: For test stability, the test method should use a data provider that accepts `(string) $text_direction`, `(array) $init_args` and `(array) $expected` parameters, should document these parameters at the bottom of the test method's docblock, and the data provider should contain a dataset for each of these tests.

File: tests/phpunit/tests/functions/mceSetDirection.php [hunk 5]
```diff
@@ -0,0 +1,51 @@
...
+
+		$wp_locale->text_direction = $direction;
```
Comment (costdev) [inline tests/phpunit/tests/functions/mceSetDirection.php:49]: This should be reset before the assertion in case of test failure.

**Review by costdev — CHANGES_REQUESTED** [review]
Thanks for the PR @pbearne! I've left some thoughts below for standards, stricter testing and test stability.

**Comment by swissspidy** [comment]
https://core.trac.wordpress.org/changeset/57718

**Pull request closed without merging**